x/lockup/keeper/lock.go

```diff
@@ -801,7 +834,7 @@ func (k Keeper) SplitLock(ctx sdk.Context, lock types.PeriodLock, coins sdk.Coin
 	splitLockID := k.GetLastLockID(ctx) + 1
 	k.SetLastLockID(ctx, splitLockID)
 
-	splitLock := types.NewPeriodLock(splitLockID, lock.OwnerAddress(), lock.Duration, lock.EndTime, coins)
+	splitLock := types.NewPeriodLock(splitLockID, lock.OwnerAddress(), lock.OwnerAddress().String(), lock.Duration, lock.EndTime, coins)
```

Comment:
This is wrong right? Shouldn't we be checking if the lock has a rewardReciever and setting it to that?

Reply:
you are correct, good catch!

Reply:
Also noticed that this method did not have unit tests, added in https://github.com/osmosis-labs/osmosis/pull/5281/commits/2d6aa3066c634f7dadc3feea7f9ce4c85dc14ae9